src/system_simulation/simulation_runner_client.rb

```diff
@@ -9,16 +9,18 @@
 class SimulationRunnerClient
   def self.update_model(json_string)
     p "server request: update_model"
-    uri = URI.parse("#{SIMULATION_RUNNER_HOST}/update_model")
-    header = {'Content-Type' => 'text/json'}
-
-    http = Net::HTTP.new(uri.host, uri.port)
-    http.read_timeout = 240
 
-    request = Net::HTTP::Post.new(uri.request_uri, header)
+    request = Sketchup::Http::Request.new("#{SIMULATION_RUNNER_HOST}/update_model", Sketchup::Http::POST)
+    request.headers = {'Content-Type' => 'text/json'}
     request.body = json_string.to_s
 
-    response = http.request(request)
+    request.start do |request, response|
+      puts "/update_model successful"
+      json_response = JSON.parse(response.body)
+      timeseries_data = parse_data(json_response["data"])
+      user_stats = json_response["user_stats"]
+      yield timeseries_data, user_stats
```

Comment:
I think I would prefer yielding only the parsed `json_response` here, since this code should only handle the networking and not the parsing (and then parsing it later)